addons/web/static/src/core/utils/autoresize.js

```diff
@@ -11,12 +11,12 @@ import { browser } from "../browser/browser";
  *
  * @param {Ref} ref
  */
-export function useAutoresize(ref, options = {}) {
+export function useAutoresize(ref, options = {}, resizeTextAreaMethod = resizeTextArea) {
```

Comment:
Not a fan of this. Your custom resizeTextArea function doesn't actually resize the textarea, but its parent. What you want is to be notified when the textarea has been resized, to be able to react and resize the parent (afaiu). You could have a custom autoresize hook that calls this one and that resizes the parent (the only duplicated part would be the add/removeEventListener, which is no big deal imo. If you really wand to modify useAutoresize, I would rather suggest to add a `onResize` option which is a callback that will be called each time the target is resized. Compared to your solution, I think that it's closer to the actual need (which is to react to a resize, not to resize an input/textarea in a custom way), the API of the hook would be better (we don't like having multiple optional arguments as it leads to weird calls) and it's complete (you only allow to customize the function for textareas, not inputs).

Don't forget that every change in the framework must be tested